### PR TITLE
VZ-7735 common-vmi code coverage

### PR DESCRIPTION
**Pull request by absBitm**

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/common/vmi_test.go [hunk 1]
```diff
@@ -152,12 +149,30 @@ func Test_FindStorageOverride(t *testing.T) {
...
+			a.NotNil(NewVMI())
+			cli := fake.NewClientBuilder().WithScheme(getScheme()).WithObjects().Build()
+			a.NoError(EnsureBackupSecret(cli))
+			a.NoError(EnsureGrafanaAdminSecret(cli))
+			a.NoError(EnsureVMISecret(cli))
+			a.NotNil(EnsureGrafanaDatabaseSecret(fakeContext))
+			a.Error(CreateOrUpdateVMI(fakeContext, nil))
+			a.False(IsVMISecretReady(fakeContext))
+			a.False(IsGrafanaAdminSecretReady(fakeContext))
+			a.Error(CreateAndLabelVMINamespaces(fakeContext))
```
Comment (adiforluls) [inline platform-operator/controllers/verrazzano/component/common/vmi_test.go:169]: Try and add the opposite cases for these funcs as well. For example, you can build the cli like `fake.NewClientBuilder().WithScheme(scheme).WithObjects(&corev1.Secret{})...`, basically inject the resource these funcs are looking for and assert accordinly.

File: platform-operator/controllers/verrazzano/component/common/vmi_test.go [hunk 2]
```diff
@@ -152,12 +149,30 @@ func Test_FindStorageOverride(t *testing.T) {
 			expectedErr: true,
 		},
 	}
+
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			a := assert.New(t)
-
 			fakeContext := spi.NewFakeContext(fake.NewClientBuilder().WithScheme(runtime.NewScheme()).Build(), &test.actualCR, nil, false, profileDir)
```
Comment (adiforluls) [inline platform-operator/controllers/verrazzano/component/common/vmi_test.go:156]: Maybe build the fake cli first and pass it to the fakeCtx

File: platform-operator/controllers/verrazzano/component/common/vmi_test.go [hunk 3]
```diff
@@ -176,6 +191,156 @@ func Test_FindStorageOverride(t *testing.T) {
 	}
 }
 
+// Test_StorageOverrideBeta1 tests the StorageOverrides functions for V1Beta1
```
Comment (adiforluls) [inline platform-operator/controllers/verrazzano/component/common/vmi_test.go:194]: Add GIVEN/WHEN/THEN style comment(s) describing the test case

**Review by adiforluls — COMMENTED** [review]
Have a few comments. There are a few code smells reported. Can you fix those as well.

File: platform-operator/controllers/verrazzano/component/common/vmi_test.go (hunk 1 above)
Comment (absBitm) [reply platform-operator/controllers/verrazzano/component/common/vmi_test.go:169]: done

File: platform-operator/controllers/verrazzano/component/common/vmi_test.go (hunk 2 above)
Comment (absBitm) [reply platform-operator/controllers/verrazzano/component/common/vmi_test.go:156]: done

**Review by adiforluls — DISMISSED** [review]

**Review by IshaGirdhar — APPROVED** [review]

**Pull request merged**